### PR TITLE
Add sc_uint to force SystemC uint type

### PR DESCRIPTION
**Pull request by b-chmiel**
There are some cases when forcing sc_uint type for certain ports is desired, for instance to prevent conversion to`bool` of signals with width 1. Similar functionality has been implemented for `sc_bv` by https://github.com/verilator/verilator/issues/402.

This PR introduces attribute `sc_uint` and Verilator pragma `/*verilator sc_uint*/` that will force emission of `AstVar` SystemC type in `AstVar::scType` to `sc_uint`.

### CODE REVIEW

**Comment by wsnyder** [comment]
There's already the --pins-sc-uint option, would it be sufficient to just have another option that also applies this to bools, or do you really absolutely need individual signal-by-signal control?

**Comment by b-chmiel** [comment]
Separate option for bools will be sufficient, signal-by-signal control is not needed. 
How such option should be called; `--pins-sc-bool-to-uint` or `--pins-sc-uint-bool`?

**Comment by wsnyder** [comment]
`--pins-sc-uint-bool` sounds good.  Appreciate the work you put into the "harder" solution though!

File: docs/guide/exe_verilator.rst [hunk 1]
```diff
@@ -1004,6 +1004,10 @@ Summary:
    :vlopt:`--pins-sc-biguint` combination, it results in sc_uint being used
    between 2 and 64 and sc_biguint being used between 65 and 512.
 
+.. option:: --pins-sc-uint-bool
+
+   Specifies SystemC inputs/outputs 1 bits wide should use sc_uint<1>.
```
Comment (wsnyder) [inline docs/guide/exe_verilator.rst:1009]:
```suggestion
   Specifies SystemC inputs/outputs one bit wide should use sc_uint<1>.
```

**Comment by b-chmiel** [comment]
Drafting, since it requires further testing. I'll get back to it next week.

**Comment by b-chmiel** [comment]
Reopening. Now, SystemC main is generated with changed types when `--pins-sc-uint-bool` is used (this also applies to `clk`).
Unfortunately, SystemC does not support `operator !` for sc_uint, therefore `clk.write(!clk.read())` workaround was used.
Also conversion uint->bool is now constrained to types with `width == 1` to avoid compilation errors.

Changes are ready for the review.

**Pull request merged**